### PR TITLE
CI update how we include docker images to support generators not part of our helm charts

### PR DESCRIPTION
**Pull request by mshidlov**
Issue Number: #3108

### Update amplication CI workflow:
1. packages to build: all packages that have a Dockerfile will be included in the image building process
2. Simplify git changes job: get all folders that have changes, against the main (master) branch on a push and against the base branch on a PR.

### Added new GitHub Actions:
1. find-folders-by - will return all folders that contain a searched file. used by the CI workflow to retrieve all folders that contain Dockerfile.
2. get-package-names - returns corresponding package names for a given list of folders. The action is to search for the package name from the package.json files. used in the CI workflow to retrieve the package names to folders that has a git diff.

### Update get-services-hash GitHub Action
Improve the hashing process and add folders to hash input instead of calculate the folders from the charts.

## PR Checklist
- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by arielweinberger** [comment]
Let's arrange a pair review on this over a call, so I can become more familiar with it please 😄

**Review by arielweinberger — CHANGES_REQUESTED** [review]

**Comment by arielweinberger** [comment]
Let's remove comments overhead :)

File: packages/amplication-data-service-generator/Dockerfile [hunk 1]
```diff
@@ -16,9 +16,9 @@ RUN npm i -g npm@7.3.0
...
 # Copy files specifying dependencies
-COPY src/server/static/package.json src/server/static/package-lock.json ./server/
-COPY src/admin/static/package.json src/admin/static/package-lock.json ./admin/
+COPY packages/amplication-data-service-generator/src/server/static/package.json packages/amplication-data-service-generator/src/server/static/package-lock.json ./server/
+COPY packages/amplication-data-service-generator/src/admin/static/package.json packages/amplication-data-service-generator/src/admin/static/package-lock.json ./admin/
 
 # Install dependencies
 RUN cd server; npm ci --silent;
-RUN cd admin; npm ci --silent;
+# RUN cd admin; npm ci --silent;
```
[the remark this answers is not in the document]
Comment (mshidlov) [reply packages/amplication-data-service-generator/Dockerfile:24]: That file will be updated in the DSG separation - the comment is only to make it run without failing I don't want to touch that file

**Review by arielweinberger — APPROVED** [review]

**Comment by arielweinberger** [comment]
Cool! Approved. It would be great to have a 30-minutes session with the team, syncing them with this change so they understand (more or less) what's new.

**Pull request closed without merging**